# Analysis of the Airlines Arrival Delay Causes

## 1)Background of the data


* The dataset is published on the open data souce Kaggle.
* It is collected by the U.S. Department of Transportation's (DOT) Bureau of Transportation Statistics (BTS). They track the on-time performance of domestic flights operated by large air carriers.
* This dataset is only for year 2008.
* Summary information on the number of on-time, delayed, canceled and diverted flights appears in DOT's monthly air Traval Consumer Report, published about 30 days after the month's end, as well as in summary tables posted on this website.
* BTS began collecting details on the causes of flight delays in June 2003. Summary statistics and raw data are made avaiable to the public at the time the Air Travle Consumer Report is released. 
* *The above information is from Kaggle. https://www.kaggle.com/giovamata/airlinedelaycauses*


## 2) Line to the Airlines Arrival Delay Dataset
* Data Introduction: https://www.kaggle.com/giovamata/airlinedelaycauses
* Data Download: https://www.kaggle.com/giovamata/airlinedelaycauses/download

In [28]:
#The direct downloadlink to the Kaggle data set
data_url='https://www.kaggle.com/giovamata/airlinedelaycauses/download'

## 3) Import the local dataset
* I've downloaded the dataset in advance to save time;
* The name of the dataset is DelayedFlights.csv

In [13]:
import pandas as pd
data_airline=pd.read_csv('DelayedFlights.csv')

## 4) Desplay the information of the dataset.
* Here, I just use the method describe of the setset to show
the maximum, minimum, mean et.al for each column

In [27]:
data_airline.head()

Unnamed: 0  Year  Month  DayofMonth  DayOfWeek  DepTime  CRSDepTime  \
0           0  2008      1           3          4   2003.0        1955   
1           1  2008      1           3          4    754.0         735   
2           2  2008      1           3          4    628.0         620   
3           4  2008      1           3          4   1829.0        1755   
4           5  2008      1           3          4   1940.0        1915   

   ArrTime  CRSArrTime UniqueCarrier  ...  TaxiIn TaxiOut  Cancelled  \
0   2211.0        2225            WN  ...     4.0     8.0          0   
1   1002.0        1000            WN  ...     5.0    10.0          0   
2    804.0         750            WN  ...     3.0    17.0          0   
3   1959.0        1925            WN  ...     3.0    10.0          0   
4   2121.0        2110            WN  ...     4.0    10.0          0   

   CancellationCode  Diverted  CarrierDelay  WeatherDelay NASDelay  \
0                 N       0.0           NaN           NaN      NaN   
1                 N       0.0           NaN           NaN      NaN   
2                 N       0.0           NaN           NaN      NaN   
3                 N       0.0           2.0           0.0      0.0   
4                 N       0.0           NaN           NaN      NaN   

  SecurityDelay  LateAircraftDelay  
0           NaN                NaN  
1           NaN                NaN  
2           NaN                NaN  
3           0.0               32.0  
4           NaN                NaN  

[5 rows x 30 columns]

In [16]:
data_airline.describe()

Unnamed: 0      Year          Month     DayofMonth      DayOfWeek  \
count  5.666450e+05  566645.0  566645.000000  566645.000000  566645.000000   
mean   8.931030e+05    2008.0       2.017748      15.615830       3.964740   
std    5.079917e+05       0.0       0.815599       8.952359       1.986924   
min    0.000000e+00    2008.0       1.000000       1.000000       1.000000   
25%    4.650050e+05    2008.0       1.000000       7.000000       2.000000   
50%    9.046460e+05    2008.0       2.000000      16.000000       4.000000   
75%    1.309138e+06    2008.0       3.000000      23.000000       6.000000   
max    1.767774e+06    2008.0       3.000000      31.000000       7.000000   

             DepTime     CRSDepTime       ArrTime     CRSArrTime  \
count  566645.000000  566645.000000  564444.00000  566645.000000   
mean     1511.366108    1458.061303    1606.59691    1622.633305   
std       450.460186     427.014137     543.35757     466.120320   
min         1.000000       0.000000       1.00000       1.000000   
25%      1156.000000    1125.000000    1312.00000    1315.000000   
50%      1539.000000    1500.000000    1708.00000    1655.000000   
75%      1858.000000    1815.000000    2027.00000    2010.000000   
max      2400.000000    2359.000000    2400.00000    2400.000000   

           FlightNum  ...       Distance         TaxiIn        TaxiOut  \
count  566645.000000  ...  566645.000000  564444.000000  566645.000000   
mean     2233.087286  ...     748.406494       6.773131      18.097412   
std      1948.828111  ...     553.648915       5.113273      13.473195   
min         1.000000  ...      24.000000       0.000000       0.000000   
25%       617.000000  ...     337.000000       4.000000      10.000000   
50%      1556.000000  ...     602.000000       6.000000      14.000000   
75%      3690.000000  ...     988.000000       8.000000      21.000000   
max      9740.000000  ...    4962.000000     189.000000     383.000000   

       Cancelled       Diverted   CarrierDelay   WeatherDelay       NASDelay  \
count   566645.0  566644.000000  375786.000000  375786.000000  375786.000000   
mean         0.0       0.003884      18.985295       3.793499      14.155022   
std          0.0       0.062203      43.017473      21.866751      32.961684   
min          0.0       0.000000       0.000000       0.000000       0.000000   
25%          0.0       0.000000       0.000000       0.000000       0.000000   
50%          0.0       0.000000       2.000000       0.000000       1.000000   
75%          0.0       0.000000      21.000000       0.000000      13.000000   
max          0.0       1.000000    1490.000000    1148.000000    1357.000000   

       SecurityDelay  LateAircraftDelay  
count  375786.000000      375786.000000  
mean        0.101310          26.020262  
std         1.908135          42.001088  
min         0.000000           0.000000  
25%         0.000000           0.000000  
50%         0.000000          10.000000  
75%         0.000000          35.000000  
max       149.000000        1254.000000  

[8 rows x 25 columns]

## 5) Display the shape of the dataset
* The shape of the data_airline is 566645*30

In [21]:
row_num,col_num=data_airline.shape
print(row_num,col_num)

566645 30


## 6) Proportion of the missing observations for each column of the dataset

In [26]:
import numpy as np
PropOfMissingValue=np.sum(pd.isnull(data_airline))/row_num
print(pd.DataFrame(PropOfMissingValue))

                          0
Unnamed: 0         0.000000
Year               0.000000
Month              0.000000
DayofMonth         0.000000
DayOfWeek          0.000000
DepTime            0.000000
CRSDepTime         0.000000
ArrTime            0.003884
CRSArrTime         0.000000
UniqueCarrier      0.000000
FlightNum          0.000000
TailNum            0.000004
ActualElapsedTime  0.003884
CRSElapsedTime     0.000176
AirTime            0.003884
ArrDelay           0.003884
DepDelay           0.000000
Origin             0.000000
Dest               0.000000
Distance           0.000000
TaxiIn             0.003884
TaxiOut            0.000000
Cancelled          0.000000
CancellationCode   0.000000
Diverted           0.000002
CarrierDelay       0.336823
WeatherDelay       0.336823
NASDelay           0.336823
SecurityDelay      0.336823
LateAircraftDelay  0.336823


## 7) Project Goal
This project is to try to find out the potential causes for the flight delay.

## 8) What is your _y_-variable?
For final project, you will need to perform a statistical model. This means you will have to accurately predict some y-variable for some combination of x-variables. From your problem statement in part 7, what is that y-variable?